docs/graphql-java-comparison.md

```diff
@@ -0,0 +1,146 @@
+
+### GraphQL Java Schema
+
+The most common way to create the schema in `graphql-java` is to first manually write the SDL file:
+
+```graphql
+schema {
+    query: Query
+}
+
+type Query {
+    bookById(id: ID): Book
+}
+
+type Book {
+    id: ID!
+    name: String!
+    pageCount: Int!
+    author: Author
+}
+
+type Author {
+    id: ID!
+    firstName: String!
+    lastName: String!
+}
+```
+
+Then write the runtime code that matches this schema to build the `GraphQLSchema` object.
+
+```java
+public class Schema {
+
+    public static void main(String[] args) {
+        SchemaParser schemaParser = new SchemaParser();
+        SchemaGenerator schemaGenerator = new SchemaGenerator();
+
+        File schemaFile = loadSchema("schema.graphqls");
+        TypeDefinitionRegistry typeRegistry = schemaParser.parse(schemaFile);
+
+        RuntimeWiring runtimeWiring = buildRuntimeWiring();
+
+        GraphQLSchema graphQLSchema = schemaGenerator.makeExecutableSchema(typeDefinitionRegistry, runtimeWiring);
+    }
+
+    private RuntimeWiring buildRuntimeWiring() {
+        return RuntimeWiring.newRuntimeWiring()
+            .type(
+                newTypeWiring("Query")
+                    .dataFetcher("bookById", graphQLDataFetchers.getBookByIdDataFetcher())
+            )
+            .type(
+                newTypeWiring("Book")
+                    .dataFetcher("author", graphQLDataFetchers.getAuthorDataFetcher())
+                    .dataFetcher("pageCount", graphQLDataFetchers.getPageCountDataFetcher())
+            )
+            .build();
+    }
+}
+
+public class GraphQLDataFetchers {
+
+    private static List<Map<String, String>> books = booksFromDB();
+    private static List<Map<String, String>> authors = authorsFromDB();
+
+    public DataFetcher getBookByIdDataFetcher() {
+        return dataFetchingEnvironment -> {
+            String bookId = dataFetchingEnvironment.getArgument("id");
+            return books
+                    .stream()
+                    .filter(book -> book.get("id").equals(bookId))
+                    .findFirst()
+                    .orElse(null);
+        };
+    }
+
+    public DataFetcher getAuthorDataFetcher() {
+        return dataFetchingEnvironment -> {
+            Map<String,String> book = dataFetchingEnvironment.getSource();
+            String authorId = book.get("authorId");
+            return authors
+                    .stream()
+                    .filter(author -> author.get("id").equals(authorId))
+                    .findFirst()
+                    .orElse(null);
+        };
+    }
+
+    public DataFetcher getPageCountDataFetcher() {
+        return dataFetchingEnvironment -> {
+            Map<String,String> book = dataFetchingEnvironment.getSource();
+            return book.get("totalPages");
+        };
+    }
+}
+```
+
+This means that there are two sources-of-truth for your schema and changes in either have to be reflected in both locations.
```

Comment:
"sources of truth", no hyphens

(Github's not letting me edit directly for some reason...)